content/methods.md

```diff
@@ -0,0 +1,416 @@
+
+## What is a Method?
+
+Methods are Meteor's remote procedure call (RPC) system, used to save user input events and data that come from the client. If you're familiar with REST APIs or HTTP, you can think of them like POST requests to your server, but with many nice features optimized for building a modern web application. Later on in this article, we'll go into detail about some of the benefits you get from Methods that you wouldn't get from an HTTP endpoint.
+
+At its core, a Method is an API endpoint for your server; you can define a Method on the server and its counterpart on the client, then call it with some data, write to the database, and get the return value in a callback. Meteor Methods are also tightly integrated with the pub/sub and data loading systems of Meteor to allow for [Optimistic UI](http://info.meteor.com/blog/optimistic-ui-with-meteor-latency-compensation) - the ability to simulate server-side actions on the client to make your app feel faster than it actually is.
+
+We'll referring to Meteor Methods with a capital M to differentiate them from class methods in JavaScript.
+
+## Defining and calling Methods
+
+### Basic Method
+
+In a basic app, defining a Meteor Method is as simple as defining a function. In a complex app, you want a few extra features to make Methods more powerful and easily testable. First, we're going to go over how to define a Method using the Meteor core API, and in a later section we'll go over how to use a helpful wrapper package we've created to enable a more powerful Method workflow.
+
+#### Defining
+
+Here's how you can use the built-in [`Meteor.methods` API](http://docs.meteor.com/#/full/meteor_methods) to define a Method. Note that Methods should always be defined in common code loaded on the client and the server to enable Optimistic UI. If you have some secret code in your Method, consult the [Security article](security.md#secret-code) for how to hide it from the client.
+
+This example uses the `aldeed:simple-schema` package, which is recommended in several other articles, to validate the method arguments.
+
+```js
+Meteor.methods({
+  'Todos.methods.updateText'({ todoId, newText }) {
+    new SimpleSchema({
+      todoId: { type: String },
+      newText: { type: String }
+    }).validate({ todoId, newText }),
+
+    const todo = Todos.findOne(todoId);
+
+    if (!todo.editableBy(this.userId)) {
+      throw new Meteor.Error('Todos.methods.updateText.unauthorized',
+        'Cannot edit todos in a private list that is not yours');
+    }
+
+    Todos.update(todoId, {
+      $set: { text: newText }
+    });
+  }
+});
+```
+
+#### Calling
+
+This method is callable from the client and server using [`Meteor.call`](http://docs.meteor.com/#/full/meteor_call). Note that you should only use a Method in the case where some code needs to be callable from the client; if you just want to modularize code that is only going to be called from the server, use a regular JavaScript function, not a Method.
```

Comment:
"a client" -- I mean server-server methods are a thing. But maybe it's better to just not mention it.